File: tests/src/main/java/com/github/klee0kai/test/tech/components/Ram.java
package com.github.klee0kai.test.tech.components;

import com.github.klee0kai.test.di.base_phone.identifiers.RamSize;

import java.util.UUID;

public class Ram {

    public final UUID uuid = UUID.randomUUID();

    public final String size;

    public Ram() {
        size = "default";
    }

    public Ram(RamSize ramSize) {
        this.size = ramSize.size;
    }
}

File: tests/src/main/java/com/github/klee0kai/test/tech/components/Ram.kt
package com.github.klee0kai.test.tech.components

import com.github.klee0kai.test.di.base_phone.identifiers.RamSize
import java.util.*

open class Ram {
    open val uuid: UUID = UUID.randomUUID()

    val size: String?

    constructor() {
        size = "default"
    }

    constructor(ramSize: RamSize?) {
        this.size = ramSize?.size
    }
}
